guides/03_features/04_padrino-cache.html.md

```diff
@@ -7,412 +7,354 @@ title: Padrino Cache
-      render 'feeds/show.iphone'
-    end
-  end
-end
-```
-
 Of course, this example assumes the markup generated by rendering
-`partials/feedcontent` would be suitable for both feed formats. This may or may
-not be the case in your application, but the principle applies: fragments are
-shared between all code which accesses the cache using the same key.
-
-### Expiring Cached Content
-
-In certain circumstances, cached content becomes stale. The `expire` helper
-removes content associated with a key or keys, which your app is then free to
-re-generate.
-
-#### expire(\*key)
-
-##### Fragment-level expiration
+`partials/feedcontent` would be suitable for both feed formats. This may or
+may not be the case in your application, but the principle applies: fragments
+are shared between all code which accesses the cache using the same key.
 
-Using the example above of a tweet server, let's suppose our users have a
-tendency to post things they quickly regret. When we query our database for new
-tweets, let's check to see if any have been deleted. If so, we'll do our user a
-favor and instantly re-render the feed.
-
-```ruby
-# Expiring fragment-level cached content
-class MyTweets < Padrino::Application
-  register Padrino::Cache
-  enable :caching
-  enable :session # we'll use this to store last time visited
-
-  COMPANY_FOUNDING = Time.utc( 2016, "April" )
-
-  controller :tweets do
-    get :feed, :map => '/:username' do
-      last_visit = session[:last_visit] || params[:since] || COMPANY_FOUNDING
-
-      username = params[:username]
-      @tweets = Tweet.since( last_visit, :username => username ).limit( 100 )
-
-      expire( "feed since #{last_visit}" ) if @tweets.any? { |t| t.deleted_since?( last_visit ) }
-
-      session[:last_visit] = Time.now
-      @feed = cache( "feed since #{last_visit}", :expires_in => 60 ) do
-        @tweets = @tweets.find_all { |t| !t.deleted? }
-        render 'partials/feedcontent'
-      end
-
-      render 'feeds/show'
-    end
-  end
-end
-```
-
-Normally, this example will only re-cache feed content every 60 seconds, but it
-will do so immediately if any tweets have been deleted.
-
-##### Page-level expiration
-
-Page-level expiration works exactly like the example above by using `expire` in
-your controller.
-
-The key is typically `env['PATH_INFO']`.
-
---------------------------------------------------------------------------------
-
-## Cache Stores
+## Caching Store
 
 You can set a global caching option or a per app caching options.
 
 ### Global Caching Options
 
-```ruby
-Padrino.cache = Padrino::Cache::Store::Memcache.new(::Memcached.new('127.0.0.1:11211', :exception_retry_limit => 1))
-Padrino.cache = Padrino::Cache::Store::Memcache.new(::Dalli::Client.new('127.0.0.1:11211', :exception_retry_limit => 1))
-Padrino.cache = Padrino::Cache::Store::Redis.new(::Redis.new(:host => '127.0.0.1', :port => 6379, :db => 0))
-Padrino.cache = Padrino::Cache::Store::Memory.new(50)
```

Comment:
Hi @wikimatze, `Memory` is now `LRUHash`. That's not particularly intuitive, so I've added a comment by each `LRUHash` reference saying it's in-memory.